plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteJdbcClient.java

```diff
@@ -0,0 +1,562 @@
+    {
+    }
+
+    @Override
+    public void renameTable(ConnectorSession session, JdbcTableHandle handle, SchemaTableName newTableName)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support rename table");
+    }
+
+    @Override
+    protected String createTableSql(RemoteTableName remoteTableName, List<String> columns, ConnectorTableMetadata tableMetadata)
+    {
+        return createTableSql(remoteTableName, columns, tableMetadata.getProperties());
+    }
+
+    private String createTableSql(RemoteTableName remoteTableName, List<String> columns, Map<String, Object> tableProperties)
+    {
+        ImmutableList.Builder<String> tableOptions = ImmutableList.builder();
+        ImmutableList.Builder<String> columnDefinitions = ImmutableList.builder();
+        columnDefinitions.addAll(columns);
+
+        List<String> primaryKeys = IgniteTableProperties.getPrimaryKey(tableProperties);
+        checkArgument(primaryKeys != null && !primaryKeys.isEmpty(), "No primary key defined for create table");
+        checkArgument(primaryKeys.size() < columns.size(), "Ignite table must have at least one non PRIMARY KEY column.");
+        columnDefinitions.add("PRIMARY KEY (" + join(", ", primaryKeys) + ")");
+
+        for (Map.Entry<String, Object> propertyEntry : tableProperties.entrySet()) {
+            String propertyKey = propertyEntry.getKey();
+            switch (propertyKey) {
+                case IgniteTableProperties.PRIMARY_KEY_PROPERTY:
+                    break;
+                case IgniteTableProperties.AFFINITY_KEY_PROPERTY:
+                    String affinityKey = IgniteTableProperties.getAffinityKey(tableProperties);
+                    if (!isNullOrEmpty(affinityKey)) {
+                        checkArgument(ImmutableSet.copyOf(primaryKeys).contains(affinityKey), "Affinity key should be one of the primary key");
+                        tableOptions.add(propertyKey + " = " + affinityKey);
+                    }
+                    break;
+                case IgniteTableProperties.BACKUPS_PROPERTY:
+                case IgniteTableProperties.TEMPLATE_PROPERTY:
+                case IgniteTableProperties.CACHE_GROUP_PROPERTY:
+                case IgniteTableProperties.CACHE_NAME_PROPERTY:
+                case IgniteTableProperties.DATA_REGION_PROPERTY:
+                case IgniteTableProperties.WRITE_SYNCHRONIZATION_MODE_PROPERTY:
+                    tableOptions.add(propertyKey + " = " + propertyEntry.getValue());
+                    break;
+                default:
+                    throw new IllegalStateException("Unexpected value: " + propertyKey);
+            }
+        }
+
+        return format("CREATE TABLE %s (%s) WITH \" %s \"", quoted(remoteTableName), join(", ", columnDefinitions.build()), join(", ", tableOptions.build()));
+    }
+
+    @Override
+    protected String quoted(@Nullable String catalog, @Nullable String schema, String table)
```

Comment:
This doesn't need to be overridden. Take a look at DruidJdbcClient for example. It also has a single catalog and schema.

Reply:
Instead of overriding `quoted` you should override `prepareQuery` (see `DruidJdbcClient#prepareTableHandleForQuery`). The `quoted` method is used in too many places some which would actually need the fully qualified name.